### ISSUE TITLE
Test Failure: failed to read response from monitor process in snapshot_scheduler_test

### ISSUE THREAD

**Issue by linh2931** [AntelopeIO/leap#1794]
Seen this again. A core dump somewhere?

https://github.com/AntelopeIO/leap/actions/runs/6555274088/job/17803647521?pr=1792#step:5:25942

**Comment by spoonincode** [comment]
https://github.com/AntelopeIO/leap/actions/runs/6568816849/job/17843972230

**Comment by heifner** [comment]
https://github.com/AntelopeIO/leap/actions/runs/6579506381/job/17875830636

**Comment by heifner** [comment]
https://github.com/AntelopeIO/leap/actions/runs/6587504928/job/17898239176

BTW, I ran the plugin_test `tests/plugin_test --catch_system_errors=no` all night locally in a loop and it never failed once.

**Comment by heifner** [comment]
https://github.com/AntelopeIO/leap/actions/runs/6591432570/job/17910300584

**Comment by spoonincode** [comment]
https://github.com/AntelopeIO/leap/actions/runs/6628104758/job/18004907600

**Comment by bhazzard** [comment]
Kevin mentioned on 10/24/23 that he ran it all night without a single failure.

**Comment by bhazzard** [comment]
Kevin also mentioned a theory that maybe something else was writing to the same directory, though most tests use a tmp directory.

**Comment by bhazzard** [comment]
move to 5.0.1 milestone

**Comment by linh2931** [comment]
The problem is shifted to read_only_trxs:
https://github.com/AntelopeIO/leap/actions/runs/6631880102/job/18016715919?pr=1819

Looks like some interferences in tests within plugin_test or in parallel test.

**Comment by spoonincode** [comment]
imo until we learn otherwise, let's consider this issue strictly for failures on `snapshot_scheduler_test` which maybe hopefully was fixed with #1821 (since there was clearly threading violations in it, crashes/weirdness isn't a stretch), so let's close it. If we should open another issue for the read only trx tests failing, let's do that.

**Comment by linh2931** [comment]
https://github.com/AntelopeIO/leap/issues/1823 opened for read only trx tests.

**Comment by linh2931** [comment]
Happened in `snapshot_scheduler_test` in this CICD run: https://github.com/AntelopeIO/leap/actions/runs/6643256190/job/18050335448?pr=1819
